zipline/data/minute_bars.py

```diff
@@ -48,6 +50,99 @@ class BcolzMinuteWriterColumnMismatch(Exception):
     pass
 
 
+class MinuteBarReader(with_metaclass(ABCMeta)):
+
+    @abstractproperty
+    def last_available_dt(self):
+        """
+        Returns
+        -------
+        dt : pd.Timestamp
+            The last minute for which the reader can provide data.
+        """
+        pass
+
+    @abstractproperty
+    def first_trading_day(self):
+        """
+        Returns
+        -------
+        dt : pd.Timestamp
+            The first trading day (session) for which the reader can provide
+            data.
+        """
+        pass
+
+    @abstractmethod
+    def get_value(self, sid, dt, field):
+        """
+        Retrieve the value at the given coordinates.
+
+        Parameters
+        ----------
+        sid : int
+            The asset identifier.
+        dt : datetime64-like
+            The minute label for the desired data point.
+        field : string
+            The OHLVC name for the desired data point.
+
+        Returns
+        -------
+        value : float|int
+            The value at the given coordinates, `float` for OHLC, `int` for
+            'volume'.
+        """
+
+    @abstractmethod
+    def get_last_traded_dt(self, asset, dt):
+        """
+        Retrieve the most recently traded minute for the given asset,
+        when viewing the data from the input minute. The last traded minute is
```

Comment:
Sounds good, will edit. And the ``dt`` when not found is `pd.NaT`. (At least that is the current implementation.)